lib/rules/no-unused-vars.js

```diff
@@ -97,6 +97,8 @@ module.exports = {
         //--------------------------------------------------------------------------
 
         var STATEMENT_TYPE = /(?:Statement|Declaration)$/;
+        var LOOP_TYPE = /^(?:DoWhile|For|ForIn|ForOf|While)Statement$/;
```

Comment:
Seems this kinda thing is needed among the rules, like - 

https://github.com/eslint/eslint/blob/b837c92cb239bbc90007118569c103c4d78c0235/lib/rules/indent.js#L616
https://github.com/eslint/eslint/blob/b837c92cb239bbc90007118569c103c4d78c0235/lib/rules/no-unmodified-loop-condition.js#L22
and so on

Can we move it in one place?